handel/src/services/stepfunctions/index.ts

```diff
@@ -0,0 +1,138 @@
+import * as deletePhasesCommon from '../../common/delete-phases-common';
+import * as deployPhaseCommon from '../../common/deploy-phase-common';
+import * as iamCalls from '../../aws/iam-calls';
+import {getTags} from '../../common/tagging-common';
+import * as handlebarsUtils from '../../common/handlebars-utils';
+import * as util from '../../common/util';
+import { HandlebarsStepFunctionsTemplate, StepFunctionsConfig } from './config-types';
+import { DeployContext, PreDeployContext, ServiceConfig, ServiceContext, UnDeployContext } from '../../datatypes';
+
+const SERVICE_NAME = 'Step Functions';
+
+/**
+ * Service Deployer Contract Methods
+ * See https://github.com/byu-oit-appdev/handel/wiki/Creating-a-New-Service-Deployer#service-deployer-contract
+ *   for contract method documentation
+ */
+
+export function check(serviceContext: ServiceContext<StepFunctionsConfig>, dependenciesServiceContexts: Array<ServiceContext<ServiceConfig>>): string[] {
+    const errors = [];
+
+    // Check that definition is a valid JSON/YAML file
+    if (!('definition' in serviceContext.params)) {
+        errors.push(`${SERVICE_NAME} - The 'definition' parameter is required.`);
+    } else if (path.extname(serviceContext.params.definition) === '.json') {
+        if (util.readJsonFileSync(serviceContext.params.definition) === null) {
+            errors.push(`${SERVICE_NAME} - ${serviceContext.params.definition} is not a valid JSON file.`);
+        }
+    } else if (['.yml', '.yaml'].includes(path.extname(serviceContext.params.definition))) {
+        if (util.readYamlFileSync(serviceContext.params.definition) === null) {
+            errors.push(`${SERVICE_NAME} - ${serviceContext.params.definition} is not a valid YAML file.`);
+        }
+    } else {
+        errors.push(`${SERVICE_NAME} - The 'definition' parameter must have file extension .json, .yml, or .yaml.`);
+    }
+
+    return errors;
+}
+
+export async function deploy(ownServiceContext: ServiceContext<StepFunctionsConfig>, ownPreDeployContext: PreDeployContext, dependenciesDeployContexts: DeployContext[]): Promise<DeployContext> {
+    const stackName = deployPhaseCommon.getResourceName(ownServiceContext);
+    winston.info(`${SERVICE_NAME} - Executing Deploy on '${stackName}'`);
+    const compiledStepFunctionsTemplate = await getCompiledStepFunctionsTemplate(stackName, ownServiceContext, dependenciesDeployContexts);
+    const stackTags = getTags(ownServiceContext);
+    const deployedStack = await deployPhaseCommon.deployCloudFormationStack(stackName, compiledStepFunctionsTemplate, [], true, SERVICE_NAME, 30, stackTags);
+    winston.info(`${SERVICE_NAME} - Finished deploying '${stackName}'`);
+    return getDeployContext(ownServiceContext, deployedStack);
+}
+
+export async function unDeploy(ownServiceContext: ServiceContext<StepFunctionsConfig>): Promise<UnDeployContext> {
+    await iamCalls.detachPoliciesFromRole(deployPhaseCommon.getResourceName(ownServiceContext));
```

Comment:
I don't think you need to do this here, since the role and policies are contained inside the CloudFormation template that will be deleted on the next line. Am I missing something though?